### PR TITLE
Prepare some api for DM

### PR DESCRIPTION
**Pull request by GuEe-GUI**
## 拉取/合并请求描述：(PR description)

[
1. Fixup:Arch Kconfig empty in MCU
2. AArch64:cpuport:replace rt_inline with macros
3. Components:MM:add early map api
4. Drivers:OFW:replace new rt_str* to string.h

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by GuEe-GUI** [comment]
@mysterywolf

File: include/rtthread.h [hunk 1]
```diff
@@ -680,6 +680,12 @@ rt_int32_t rt_memcmp(const void *cs, const void *ct, rt_size_t count);
 #endif /* RT_KSERVICE_USING_STDLIB_MEMORY */
 char *rt_strdup(const char *s);
 rt_size_t rt_strnlen(const char *s, rt_ubase_t maxlen);
+char *rt_strchr(const char *s, int c);
```
Comment (BernardXiong) [inline include/rtthread.h:683]: 不建议无限期的扩大化了

Comment (mysterywolf) [reply include/rtthread.h:683]: 可以考虑加上，有RT_KSERVICE_USING_STDLIB罩着，这个宏正常情况都是关闭的

Comment (mysterywolf) [reply include/rtthread.h:683]: 确实，这块很容易可能无限扩充下去，这个本应该是libc的责任。

@GuEe-GUI 如果这些函数标准C库可以直接使用的话，直接#include <string.h>吧

File: libcpu/aarch64/Kconfig [hunk 2]
```diff
@@ -1,5 +1,7 @@
 if ARCH_ARMV8 && ARCH_CPU_64BIT
 
+menu "RT-Thread Architecture"
```
Comment (mysterywolf) [inline libcpu/aarch64/Kconfig:3]:
```suggestion
menu "AArch64 Architecture Configuration"
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: components/drivers/ofw/irq.c [hunk 3]
```diff
@@ -130,7 +131,7 @@ static rt_err_t ofw_parse_irq_map(struct rt_ofw_node *np, struct rt_ofw_cell_arg
      *              <0x1800 0 0   4 &gic    0 0 GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>; // INTD
      *      };
      *
-     * In fact, basically no SoC will be use multi ic to implemented INTx.
+     * In fact, amost no SoC will be use multi ic to implemented INTx.
```
Comment (mysterywolf) [inline components/drivers/ofw/irq.c:134]: 单词拼写有误

Comment (GuEe-GUI) [reply components/drivers/ofw/irq.c:134]: almost？

**Review by mysterywolf — APPROVED** [review]

**Comment by GuEe-GUI** [comment]
@BernardXiong

**Review by mysterywolf — APPROVED** [review]

**Review by BernardXiong — APPROVED** [review]

**Pull request merged**